src/tls/rustls.rs

```diff
@@ -66,6 +73,39 @@ impl TlsStream<TokioTcpStream> {
         TlsStream::with_connector(con, url).await
     }
 
+    /// Create a new TLS connection over TCP using native certificates.
+    ///
+    /// Unlike [`TlsStream::connect`], creates a root certificates store populated with the certificates
+    /// loaded from a platform-native certificate store. Thos method also allows to ***dangerously***
+    /// skip server certificates verification.
+    pub async fn connect_with_native_certs(
```

Comment:
What's the rationale for us providing this rather than just providing a `connect_with_tls_config` that takes a `ClientConfig` directly which the user can configure however they would like? Or alternatively just telling users to use `TlsStream::new`/`TlsStream::with_connector` directly?

Reply:
Well, I've saw they are doing so in the [`imap`](https://github.com/jonhoo/rust-imap/blob/8f3e678ee33caf39c6ec6135a2a79b192ca56620/src/client_builder.rs#L64-L72) crate, and I loved the idea of giving the user such a shortcut.

We ourselves making use of this `TlsStream::connect_with_native_certs` in the `WokrerBuilder` terminal method. Just handy, wdyt?